language/java/mistakes.go

```diff
@@ -0,0 +1,55 @@
+package java
+
+import (
+	"context"
+	"regexp"
+	"sort"
+
+	"golang.org/x/exp/maps"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/symflower/eval-dev-quality/util"
+)
+
+// Mistakes builds a Java repository and returns the list of mistakes found.
+func Mistakes(logger *log.Logger, repositoryPath string) (mistakes []string, err error) {
+	output, err := util.CommandWithResult(context.Background(), logger, &util.Command{
+		Command: []string{
+			"mvn",
+			"clean",
+			"install",
+		},
+
+		Directory: repositoryPath,
+	})
+	if err != nil {
+		if output != "" {
+			return extractMistakes(output), nil
+		}
+
+		return nil, pkgerrors.WithStack(err)
+	}
+
+	return nil, nil
+}
+
+// mistakesRe defines the structure of a Java mistake.
+var mistakesRe = regexp.MustCompile(`(?m)\[ERROR\] (.*:\[\d+,\d+\].*)$`)
+
+func extractMistakes(rawMistakes string) (mistakes []string) {
+	m := map[string]bool{}
```

Comment:
Why a map? Can there be multiple entries with the same error message?

Reply:
You can check the log in `TestExtractMistakes`,  the same errors are reported after the `COMPILATION ERROR` and after `BUILD FAILURE` logs